lib/main.dart

```diff
@@ -1,45 +1,156 @@
+import 'package:bagzz/custom_icon/custom_icon_icons.dart';
+import 'package:bagzz/models/bags_slides.dart';
+import 'package:dot_navigation_bar/dot_navigation_bar.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
 import 'package:flutter_svg/flutter_svg.dart';
 
 void main() {
-  runApp(const MaterialApp(
+  runApp(MaterialApp(
     home: MyApp(),
   ));
 }
 
-class MyApp extends StatelessWidget {
-  const MyApp({Key? key}) : super(key: key);
+class MyApp extends StatefulWidget {
+  MyApp({Key? key}) : super(key: key);
 
+  @override
+  State<MyApp> createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
   @override
   Widget build(BuildContext context) {
+    num devicePixel = MediaQuery.of(context).devicePixelRatio;
+    final PageController controller =
+        PageController(initialPage: 0, keepPage: true);
```

Comment:
@edoliantesjr Initializing controller without disposing will caused memory leak.
don't forget to dispose the controller during widget dispose

call:
 @override
  void dispose(){
    controller.dispose();
    super.dispose();
  }